content/error-handling/error-codes.textile

```diff
@@ -0,0 +1,301 @@
+“device”: “phone”
+```
+
+The @id@ property is invalid because it is a numeric value instead of the required string.
+
+*Resolution:*
+- The @id@ property must be a string. Ensure the request includes a valid string value for the @id@ property, such as:
+
+```
+{
+“id”: “12345”,
+“platform”: “android”,
+“device”: “phone”
+}
+```
+
+h3(#40022). 40022 - Invalid resource
+
+The requested resource is invalid or already exists.
+
+An example of this error is attempting to create an app named "MyApp" when an app with that name already exists or attempting to create a user account with an email address that is already registered.
+
+*Resolution:* Ensure unique app's name has valid capabilities in requests.
+
+h2(#connection). Connection errors
+
+h3(#80000). 80000 - Connection failed
+
+The SDK is failing to connect to Ably, which is likely due to issues with client-side network connectivity.
+
+An example of this error is a client application using the SDK fails to connect to Ably due to network issues, such as being offline or blocked by a firewall.
+
+*Resolution:* No immediate action is required, as the SDK will automatically retry the "connection":/connect#connection-states after 15 seconds.
+
+h3(#80002). 80002 - Connection suspended
+
+The SDK cannot connect to Ably, likely due to client-side network issues, and has failed to establish a connection within 2 minutes.
+
+An example of this error is the client application loses network connectivity, and after 2 minutes of failed retries, the connection transitions to the suspended state, indicating prolonged unavailability.
+
+*Resolution:* Check your network connection and verify that the client device has stable internet access. Retry the "connection":/error-handling#connections and call the @.connect()@ method to reattempt a connection manually.
+
+h3(#80003). 80003 - Generic disconnection error
+
+The SDK has trouble connecting to Ably, likely due to client-side network connectivity issues. The specific cause cannot be determined.
+
+An example of this error is a client application loses its connection to Ably due to an unspecified issue, such as Wi-Fi dropouts. The detailed logs or error messages do not explain the exact cause.
+
+*Resolution:* No immediate action is required, as the SDK will automatically retry the connection after 15 seconds. Check your network connectivity and verify that the client device has stable internet access. Retry the connection and call the @.connect()@ method to reattempt a connection manually.
+
+h3(#80018). 80018 - Invalid connection ID (invalid format)
+
+An invalid connection ID was supplied.
+
+An example of this error is an invalid connection ID is when the client application sends a message with a connection ID that doesn't match the current connection state due to an outdated SDK version.
+
+*Resolution:* Upgrade to the latest version of the SDK to ensure that the connection lifecycle is properly managed and that valid connection IDs are used.
```

Comment:
This is too specific to that 1 version of the JS library to be an example / resolution for all occurrences.